### PR TITLE
ui: improve display of editor, brief and detail views

### PR DESCRIPTION
**Pull request by AoNoOokami**
* Adds horizontal wrapper to move editor labels next to the input.
* Compacts brief and detail views to not use the whole width of the window.
* Displays description in tooltips in editor.

Co-Authored-by: Alicia Zangger <alicia.zangger@rero.ch>

## Why are you opening this PR?

Implements task 1473 of US1296

https://tree.taiga.io/project/rero21-reroils/us/1296?milestone=263424
https://tree.taiga.io/project/rero21-reroils/task/1473?kanban-status=1224894

## How to test?

Use https://github.com/rero/rero-ils/pull/983 and https://github.com/rero/rero-ils-ui/pull/267

Use `npm pack` or `npm link` to install this PR on rero-ils-ui, which needs to be installed in rero-ils after that.
Go to document editor and check that the labels are next to the inputs (same line, not above).

## Code review check list

- [ ] Commit message template compliance.
- [ ] Commit message without typos.
- [ ] File names.
- [ ] Functions names.
- [ ] Functions docstrings.
- [ ] Unnecessary commited files?
- [ ] Extracted translations?

### CODE REVIEW

File: projects/rero/ng-core/src/lib/record/detail/detail.component.html [hunk 1]
```diff
@@ -14,27 +14,29 @@
...
-            {{ 'Delete' | translate }}
-        </button>
-        <ng-template #deleteMessageLink>
-            <button class="btn btn-sm btn-dark ml-1" [title]="'Delete'|translate"
-                (click)="showDeleteMessage(deleteStatus.message)" *ngIf="deleteStatus.message">
-                <i class="fa fa-trash mr-1"></i>{{ 'Delete' | translate }}
-            </button>
-        </ng-template>
-    </ng-container>
+<div class="container">
```
Comment (sebastiendeleze) [inline projects/rero/ng-core/src/lib/record/detail/detail.component.html:17]: I think it's a bad idea to wrap content into a .container. Layout is not the responsability of ng-core and for example, I don't want this behavior on SONAR.

File: projects/rero/ng-core/src/lib/record/editor/array-type/array-type.component.html [hunk 2]
```diff
@@ -15,12 +15,12 @@
...
 -->
 <!-- array title only if the array is empty -->
-<label *ngIf="to.label && to.hideLabel !== true && formControl && formControl.length === 0" [attr.for]="id">
+<label *ngIf="to.label && to.hideLabel !== true && formControl && formControl.length === 0" [attr.for]="id" tooltip="{{ to.description }}">
   <!-- add button only if the max is not reached -->
   <a *ngIf="canAdd()" (click)="add(0)" class="btn btn-link btn-sm">
     <i class="fa fa-plus"></i>
   </a>
-  <span [tooltip]="to.description">{{ to.label }}</span>
+  <span tooltip="{{to.description}}">{{ to.label }}</span>
```
Comment (sebastiendeleze) [inline projects/rero/ng-core/src/lib/record/editor/array-type/array-type.component.html:23]: Why this change? Double moustache is the preferred way for binding attributes?

File: projects/rero/ng-core/src/lib/record/search/record-search.component.html [hunk 3]
```diff
@@ -31,7 +31,7 @@
       </a>
     </li>
   </ul>
-  <div *ngIf="!isLoading; else loading">
+  <div *ngIf="!isLoading; else loading" class="container ">
```
Comment (sebastiendeleze) [inline projects/rero/ng-core/src/lib/record/search/record-search.component.html:34]: Same here, *.container* must not be here.

File: projects/rero/ng-core/src/lib/record/detail/detail.component.html (hunk 1 above)
Comment (AoNoOokami) [reply projects/rero/ng-core/src/lib/record/detail/detail.component.html:17]: Done following your advices

File: projects/rero/ng-core/src/lib/record/search/record-search.component.html (hunk 3 above)
Comment (AoNoOokami) [reply projects/rero/ng-core/src/lib/record/search/record-search.component.html:34]: Done following your advices

File: projects/rero/ng-core/src/lib/record/editor/array-type/array-type.component.html (hunk 2 above)
Comment (AoNoOokami) [reply projects/rero/ng-core/src/lib/record/editor/array-type/array-type.component.html:23]: I thought I was following the right syntax but both are possible.

File: projects/rero/ng-core/src/lib/record/detail/detail.component.html [hunk 4]
```diff
@@ -14,27 +14,29 @@
...
-            {{ 'Delete' | translate }}
-        </button>
-        <ng-template #deleteMessageLink>
-            <button class="btn btn-sm btn-outline-danger ml-1 disabled" [title]="'Delete'|translate"
-                (click)="showDeleteMessage(deleteStatus.message)" *ngIf="deleteStatus.message">
-                <i class="fa fa-trash mr-1"></i>{{ 'Delete' | translate }}
-            </button>
-        </ng-template>
-    </ng-container>
+<div class="ui-content">
```
Comment (sebastiendeleze) [inline projects/rero/ng-core/src/lib/record/detail/detail.component.html:17]: I don't like `ui-content` because all the page is a user interface, not only the content. Please change to one of these: 
* `<div class="main-content">`
* `<div class="main-section">`
* `<div class="main">`

File: projects/rero/ng-core/src/lib/record/detail/detail.component.html [hunk 5]
```diff
@@ -14,27 +14,29 @@
...
+  <div class="alert alert-danger" *ngIf="error">{{ error | translate }}</div>
+  <div class="float-right ml-4 mt-2 mb-4" *ngIf="record && adminMode.can">
+      <a href class="btn btn-sm btn-primary" routerLink="../../edit/{{ record.metadata.pid }}"
```
Comment (jma) [inline projects/rero/ng-core/src/lib/record/detail/detail.component.html:20]: I would change routerLink as:
```
    [routerLink]="['../../edit', record.metadata.pid]"
```

File: projects/rero/ng-core/src/lib/record/editor/editor.component.ts [hunk 6]
```diff
@@ -284,6 +287,13 @@ export class EditorComponent implements OnInit, OnDestroy {
             ];
           }
 
+          // Add an horizontal wrapper
+          if (this._horizontalWrapperTypes.indexOf(field.type) >= 0) {
```
Comment (jma) [inline projects/rero/ng-core/src/lib/record/editor/editor.component.ts:291]: I would use some here.

File: projects/rero/ng-core/src/lib/record/editor/horizontal-wrapper/horizontal-wrapper.component.ts [hunk 7]
```diff
@@ -0,0 +1,41 @@
+/*
+ * RERO angular core
```
Comment (jma) [inline projects/rero/ng-core/src/lib/record/editor/horizontal-wrapper/horizontal-wrapper.component.ts:2]: Bad licence.

File: projects/rero/ng-core/src/lib/record/search/record-search.component.html [hunk 8]
```diff
@@ -31,7 +31,7 @@
       </a>
     </li>
   </ul>
-  <div *ngIf="!isLoading; else loading">
+  <div *ngIf="!isLoading; else loading" class="ui-content">
```
Comment (jma) [inline projects/rero/ng-core/src/lib/record/search/record-search.component.html:34]: If you change the ui-content class as suggest by Sebatien do not forget this.

File: projects/rero/ng-core/src/lib/record/editor/horizontal-wrapper/horizontal-wrapper.component.ts (hunk 7 above)
Comment (AoNoOokami) [reply projects/rero/ng-core/src/lib/record/editor/horizontal-wrapper/horizontal-wrapper.component.ts:2]: Which one should I use?

File: projects/rero/ng-core/src/lib/record/editor/editor.component.ts (hunk 6 above)
Comment (AoNoOokami) [reply projects/rero/ng-core/src/lib/record/editor/editor.component.ts:291]: Some what?
